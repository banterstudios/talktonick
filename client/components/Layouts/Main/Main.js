import React from 'react'
import PropTypes from 'prop-types'
import Wrapper from 'client/components/Gui/Wrapper'
import Header from 'client/components/Common/Header'
import glamorous from 'glamorous'

const Content = glamorous.div(({ theme: { headerHeight } }) => ({
  position: 'relative',
  width: '100%',
  paddingTop: headerHeight
}))

const Main = ({ children }) => {
  return (
    <Wrapper>
      <Header />
      <Content>
        { children }
      </Content>
    </Wrapper>
  )
}

Main.PropTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.node
  ])
}

export default Main
