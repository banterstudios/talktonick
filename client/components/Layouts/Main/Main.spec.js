import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Main from './Main'

describe('components/Layouts/Main', () => {
  describe('@renders', () => {
    it('should render correctly', () => {
      const component = shallow(<Main />)

      expect(toJson(component)).toMatchSnapshot()
      component.unmount()
    })
  })
})
