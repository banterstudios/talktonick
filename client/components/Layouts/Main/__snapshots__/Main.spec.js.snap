// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layouts/Main @renders should render correctly 1`] = `
<glamorous(section)>
  <Header />
  <glamorous(div) />
</glamorous(section)>
`;
